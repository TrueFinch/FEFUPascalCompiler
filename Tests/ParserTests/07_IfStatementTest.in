program a;
var
    i: integer;
begin
    i := 1;
    if (i > 0) then
    begin
        i += 2;
        write(i);
    end
    else
    begin
        i -= 2;
        write(i);
    end;
end.