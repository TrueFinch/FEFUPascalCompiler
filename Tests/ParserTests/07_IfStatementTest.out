                                 a                                  
                                 ┼                                  
                             MainBlock                              
       ┌─────────────────────────┴──────┐                           
      var                       CompoundStatement                   
       ┼         ┌──────────────────────┴──┐                        
 SimpleVarDecl   :=                   IfStatement                   
 ┌─────┼        ┌┴─┐   ┌────────────┬──────┴────────────┐           
 i  integer     i  1   >    CompoundStatement   CompoundStatement   
                      ┌┴─┐   ┌──────┴──┐         ┌──────┴──┐        
                      i  0   +=   FunctionCall   -=   FunctionCall  
                            ┌┴─┐    ┌──┴─┐      ┌┴─┐    ┌──┴─┐      
                            i  2  write  i      i  2  write  i      